Source/JavaScriptCore/inspector/InjectedScriptSource.js

```diff
@@ -207,14 +207,16 @@ let InjectedScript = class InjectedScript extends PrototypelessObjectBase
         return this._evaluateAndWrap(callFrame.evaluateWithScopeExtension, callFrame, expression, objectGroup, isEvalOnCallFrame, includeCommandLineAPI, returnByValue, generatePreview, saveResult);
     }
 
-    callFunctionOn(objectId, expression, args, returnByValue, generatePreview)
+    callFunctionOn(objectId, expression, args, returnByValue, generatePreview, awaitPromise, callback)
     {
         let parsedObjectId = this._parseObjectId(objectId);
         let object = this._objectForId(parsedObjectId);
         let objectGroupName = this._idToObjectGroupName[parsedObjectId.id];
 
-        if (!isDefined(object))
-            return "Could not find object with given id";
+        if (!isDefined(object)) {
+            callback(this._createThrownValue("Could not find object with given id", objectGroupName));
```

Comment:
i think this is actually a different behavior in that the protocol command would work (but return a synthetic error object) instead of erroring (with this string)

Reply:
Yes, you are right. Updated this and the other place and added a couple tests.